daemon/emer-main.c

```diff
@@ -100,6 +100,91 @@ on_set_enabled (EmerEventRecorderServer *server,
   return TRUE;
 }
 
+static gboolean
+on_start_aggregate_timer (EmerEventRecorderServer *object,
+                          GDBusMethodInvocation   *invocation,
+                          GVariant                *event_id,
+                          GVariant                *aggregate_key,
+                          gboolean                 has_payload,
+                          GVariant                *payload,
+                          EmerDaemon              *daemon)
```

Comment:
If an app calls this D-Bus method, and then disconnects from the bus without calling Stop (e.g. because it crashe), the timer will run forever.

The fix would be to use [g_bus_watch_name_on_connection](https://gnome.pages.gitlab.gnome.org/gtk/gio/func.bus_watch_name_on_connection.html) (connection, sender, G_BUS_NAME_WATCHER_FLAGS_NONE, NULL, vanished_cb, ...), stop the timer when the callback fires, and remove the watch when the timer is stopped.

This could be a follow-up branch but it does need to happen IMO.

Reply:
I don't see a need for checking polkit connections for clients in this case.

Reply:
I opened https://phabricator.endlessm.com/T32641 for this.

It is not a regression that gnome-shell crashing means running apps' time open so far is lost – that was true in the old stack. But now there is dangling state in the daemon, I think it's more important to clean up.